### PR TITLE
fix: Minor fix allowing partition changes as per main.tf

### PR DESCRIPTION
**Pull request by ilunam**
## Description
Minor fix allowing partition changes [as per main.tf](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/5955d26589979abdad832e4e9a2dec9753ebc738/main.tf#L227).

## Motivation and Context
Failed when using a different partition.

## Breaking Changes
No.

## How Has This Been Tested?
Unsure how to test it in example context. Require differing partitions.

I've run the code against my scenario and it worked fine.

### CODE REVIEW

File: node_groups.tf [hunk 1]
```diff
@@ -30,7 +30,7 @@ data "aws_iam_policy_document" "cni_ipv6_policy" {
   statement {
     sid       = "CreateTags"
     actions   = ["ec2:CreateTags"]
-    resources = ["arn:aws:ec2:*:*:network-interface/*"]
+    resources = ["arn:${data.aws_partition.current.dns_suffix}:ec2:*:*:network-interface/*"]
```
Comment (bryantbiggs) [inline node_groups.tf:33]:
```suggestion
    resources = ["arn:${data.aws_partition.current.partition}:ec2:*:*:network-interface/*"]
```

**Comment by bryantbiggs** [comment]
closing for now

**Comment by psviderski** [comment]
@bryantbiggs Why was it closed and not merged with the suggested change applied? Would you like a new PR to be created with `data.aws_partition.current.partition`?

**Comment by bryantbiggs** [comment]
Because it was incorrect as written and hadn't been updated in some time

**Pull request closed without merging**